From 49772dd2b2b42e2be8e50b6fba51739b8a36eb10 Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Wed, 19 Jun 2013 15:43:49 +0800
Subject: [PATCH 16/39] ENGR00267092 Alarm can't open if choosing it from
 StatusBar's quick setting

Change the package to be started aligned with the default DeskClock package

Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
---
 .../src/com/android/systemui/statusbar/phone/QuickSettings.java     | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/phone/QuickSettings.java b/packages/SystemUI/src/com/android/systemui/statusbar/phone/QuickSettings.java
index e7b8fa1..b40916f 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/phone/QuickSettings.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/phone/QuickSettings.java
@@ -658,6 +658,12 @@ class QuickSettings {
             @Override
             public void onClick(View v) {
                 startSettingsActivity(AlarmClock.ACTION_SHOW_ALARMS);
+                // TODO: Jump into the alarm application
+                Intent intent = new Intent();
+                intent.setComponent(new ComponentName(
+                        "com.android.deskclock",
+                        "com.android.deskclock.AlarmClock"));
+                startSettingsActivity(intent);
             }
         });
         mModel.addAlarmTile(alarmTile, new QuickSettingsModel.RefreshCallback() {
-- 
1.8.0

